nrf9120 support

nerwalt opened the pull request:
Adding support for the nrf9120.  This is the internal part number for the nrf91*1 series SiPs.

tysonl-trace committed:
Initial commit on nrf9151 preview
Updated nordic nrfxlib to v2.7.0-rc2
Created features for the nrf9160, nrf9151, nrf9161, and nrf9120. The
nrf9120 is the internal part number for the nrf9161 and nrf9151, so it
can be used for either of those chips

tysonl-trace committed:
Adding nrf9161 feature

tysonl-trace committed:
Updated nordic nrfxlib to v2.7.0

tysonl-trace committed:
Making nrf9160 default

tysonl-trace committed:
Setting version to 2.7.0 to match underlying nrfxlib version

nerwalt committed:
Fix cargo fmt

diondokter commented on Cargo.toml:
```diff
@@ -1,13 +1,13 @@
 [package]
 name = "nrfxlib-sys"
-version = "2.4.2"
+version = "2.7.0"
```
This is technically not correct since this is a breaking change update. But it is consistent with what we've been doing in this crate.

Could you put in the readme in the 'Using' section that this crate does not follow semver and that you should pin the crate to an exact version? (Oh and update the version mentioned in the readme)

At the same time you can add to the changelog in the readme.

nerwalt replied:
I looked at the version history and just guessed that this crate's version tracked the nordic library versions.

diondokter commented on Cargo.toml:
```diff
@@ -40,6 +40,12 @@ bindgen = "0.66.1"
 arm-none-eabi-objcopy = []
 llvm-objcopy = ["dep:llvm-tools"]
+
+nrf9160 = []
+nrf9151 = ["nrf9120"]
+nrf9161 = ["nrf9120"]
+nrf9120 = []
```
Could you mention these in the readme too so people know it supports the multiple versions?

diondokter commented on build.rs:
```diff
@@ -78,8 +78,18 @@ fn main() {
 	let bindings_out_path = PathBuf::from(env::var("OUT_DIR").unwrap()).join("bindings.rs");
 	std::fs::write(bindings_out_path, rust_source).expect("Couldn't write updated bindgen output");
 
+	// The nrf9160 feature is on by default for legacy support, so we check for the nrf9120 feature
+	// first to handle the case in which both nrf9120 and nrf9160 are set (i.e., nrf9120 is set,
+	// but no-defaults is not set)
```
This seems a bit weird... Having both 9160 and 9120 on at the same time should be a compile error too

nerwalt replied:
Agreed.  I went back and forth on this.  I was aiming to backward compatible.

diondokter commented on build.rs:
```diff
@@ -132,9 +142,9 @@ fn main() {
 		"cargo:rustc-link-search={}",
 		Path::new(&nrfxlib_path)
-			.join("crypto/nrf_oberon/lib/cortex-m33/hard-float")
+			.join("crypto/nrf_oberon/lib/cortex-m33/soft-float")
```
Why soft float? We have an FPU, so why not use it?

diondokter replied:
You resolved this review comment, but it's still soft float...

nerwalt replied:
Sorry.  I reverted the modem library but not the oberon library.

nerwalt committed:
Updating README
Removing nrf9160 as default. User must select a chip.
Using hard float libarary build

nerwalt committed:
Updated Using section of README
Adding note on selecting a chip feature

diondokter commented:
I'll try to find some time to test if everything still works combined with your PR for nrf-modem.
Might be this thursday

diondokter commented:
Alright tested and it works fine. If you make the CI pass I'll merge and release. Same for the nrf-modem PR

nerwalt committed:
Using hard-float version of oberon

nerwalt committed:
cargo fmt

nerwalt committed:
Fixing CI
cargo clippy is now run for the two major chip versions
